src/lib/autoinstall/clients/autoyast.rb

```diff
@@ -0,0 +1,196 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "autoinstall/auto_sequence"
+
+Yast.import "Pkg"
+Yast.import "Wizard"
+Yast.import "Mode"
+Yast.import "Profile"
+Yast.import "AutoinstConfig"
+Yast.import "Y2ModuleConfig"
+Yast.import "Popup"
+Yast.import "AddOnProduct"
+Yast.import "CommandLine"
+Yast.import "AutoInstall"
+Yast.import "UI"
+
+module Y2Autoinstallation
+  module Clients
+    # This client is responsible for starting the AutoYaST UI.
+    class Autoyast < Yast::Client
+      include Yast::Logger
+
+      def initialize
+        textdomain "autoinst"
+
+        Yast::Mode.SetMode("autoinst_config")
+
+        Yast.include self, "autoinstall/dialogs.rb"
+        Yast.include self, "autoinstall/conftree.rb"
+      end
+
+      def main
+        if Yast::Y2ModuleConfig.GroupMap.empty?
+          Yast::Wizard.CreateDialog
+          Yast::Popup.Error(_("Error while reading configuration data."))
+          Yast::Wizard.CloseDialog
+          return :abort
+        end
+
+        turn_off_signature_checks
+
+        @cmdline = {
+          "id"         => "autoyast",
+          "help"       => _("AutoYaST"),
+          "guihandler" => fun_ref(method(:auto_sequence), "any ()"),
+          "actions"    => {
+            "file"         => {
+              "handler" => fun_ref(
+                method(:run_ui),
+                "boolean (map <string, string>)"
+              ),
+              "help"    => "file operations"
```

Comment:
I think we need to improve also help texts, it does not make sense here. BTW I would mark both file and module as deprecated and use just something like "run"